lib/ble/local.toit

```diff
@@ -59,28 +58,55 @@ class Peripheral extends Resource_:
   */
   start-advertise
       data/AdvertisementData
+      --scan-response/AdvertisementData?=null
       --interval/Duration=DEFAULT-INTERVAL
       --connection-mode/int=BLE-CONNECT-MODE-NONE:
-    if platform == system.PLATFORM-MACOS:
+    if system.platform == system.PLATFORM-MACOS:
```

Comment:
Looks like `scan-response` is ignored on macOS. Should we throw if you pass a non-null value?

Reply:
I think we could just merge the advertisements and let macos deal with it. Something to test/play with in a follow-up PR.
I also added checks for unsupported fields.
done.